pymc3/variational/advi_minibatch.py

```diff
@@ -203,53 +216,167 @@ def logp_(z_g):
-                   encoder_params=[], total_size=None, optimizer=None,
-                   learning_rate=.001, epsilon=.1, random_seed=None,
-                   mode=None):
-    """Perform mini-batch ADVI.
-
-    This function implements a mini-batch ADVI with the meanfield
-    approximation. Autoencoding variational inference is also supported.
-
-    The log probability terms for mini-batches, corresponding to RVs in
-    minibatch_RVs, are scaled to (total_size) / (the number of samples in each
-    mini-batch), where total_size is an argument for the total data size.
-
-    minibatch_tensors is a list of tensors (can be shared variables) to which
-    mini-batch samples are set during the optimization. In most cases, these
-    tensors are observations for RVs in the model.
+def _make_elbo_t(
+    observed_RVs, global_RVs, local_RVs, potentials, n_mcsamples, random_seed):
+    global_order = pm.ArrayOrdering([v for v in global_RVs])
+    local_order = pm.ArrayOrdering([v for v in local_RVs])
 
-    local_RVs and observed_RVs are used for autoencoding variational Bayes.
-    Both of these RVs are associated with each of given samples.
-    The difference is that local_RVs are unkown and their posterior
-    distributions are approximated.
+    inarray_g, uw_g, replace_g, c_g = _join_global_RVs(global_RVs, global_order)
+    inarray_l, uw_l, replace_l, c_l = _join_local_RVs(local_RVs, local_order)
 
-    local_RVs are Ordered dict, whose keys and values are RVs and a tuple of
-    two objects. The first is the theano expression of variational parameters
-    (mean and log of std) of the approximate posterior, which are encoded from
-    given samples by an arbitrary deterministic function, e.g., MLP. The other
-    one is a scaling constant to be multiplied to the log probability term
-    corresponding to the RV.
+    logpt = _make_logpt(global_RVs, local_RVs, observed_RVs, potentials)
+    replace = replace_g
+    replace.update(replace_l)
+    logpt = theano.clone(logpt, replace, strict=False)
 
-    observed_RVs are also Ordered dict with RVs as the keys, but whose values
-    are only the scaling constant as in local_RVs. In this case, total_size is
-    ignored.
+    elbo = _elbo_t(logpt, uw_g, uw_l, inarray_g, inarray_l, c_g, c_l,
+                   n_mcsamples, random_seed)
 
-    If local_RVs is None (thus not using autoencoder), the following two
-    settings are equivalent:
+    return elbo, uw_l, uw_g
 
-    - observed_RVs=OrderedDict([(rv, total_size / minibatch_size)])
-    - minibatch_RVs=[rv], total_size=total_size
 
-    where minibatch_size is minibatch_tensors[0].shape[0].
+def advi_minibatch(vars=None, start=None, model=None, n=5000, n_mcsamples=1,
+                   minibatch_RVs=None, minibatch_tensors=None,
+                   minibatches=None, global_RVs=None, local_RVs=None,
+                   observed_RVs=None, encoder_params=[], total_size=None,
```

Comment:
mutable default value seems to be modified [here](https://github.com/pymc-devs/pymc3/pull/1573/files#diff-1e408f9f0670474e10d1d854282138d0R521)
You can use
```python
def advi_minibatch(
    ...
    observed_RVs=None, encoder_params=None, total_size=None
    ...
    )
    if encoder_params is None:
        encoder_params = []
```